src/components/scene/background.js

```diff
@@ -15,6 +16,30 @@ module.exports.Component = register('background', {
       return;
     }
     object3D.background = new THREE.Color(data.color);
+
+    if (data.generateEnvironment) {
```

Comment:
Maybe we can separate into a separate `initEnvironmentMap` method. I think we just need to run it once and not recreate if the component is updated. We should check if there's already an environment.

Reply:
Thanks for the spot this should've gone in init() not update!!